### PR TITLE
Autolev parser docs 3

### PR DESCRIPTION
**Pull request by NikhilPappu**
- other
  - Added the issues and future improvements sections to autolev_parser.rst in the mechanics part of the documentation.

### CODE REVIEW

**Comment by certik** [comment]
@NikhilPappu is this ready to be merged? It looks good to me.

File: sympy/utilities/runtests.py [hunk 1]
```diff
@@ -662,6 +662,7 @@ def _doctest(*paths, **kwargs):
     split  = kwargs.get('split', None)
     blacklist.extend([
         "doc/src/modules/plotting.rst",  # generates live plots
+        "doc/src/modules/physics/mechanics/autolev_parser.rst",
```
Comment (certik) [inline sympy/utilities/runtests.py]: Why is this blacklisted?

Comment (NikhilPappu) [reply sympy/utilities/runtests.py]: The doctest in autolev_parser.rst is not meant to be executed. It just shows the user what commands should be executed in the shell.

**Comment by certik** [comment]
Thanks!

File: doc/src/modules/physics/mechanics/autolev_parser.rst [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+
+     Autolev Code
+     ------------
+     SOLVE(ZERO,X,Y)
+     A = RHS(X)*2 + RHS(Y)
+
+  .. code-block:: python
+
+     SymPy Code
+     ----------
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/autolev_parser.rst]: These should be made comments with `#`.

**Pull request merged**